SConstruct

```diff
@@ -1174,42 +1174,142 @@ for libraryName, libraryDef in libraries.items() :
 # Graphics
 #########################################################################################################
 
-def buildGraphics( target, source, env ) :
-
-	svgFileName = os.path.abspath( str( source[0] ) )
-
-	dir = os.path.dirname( os.path.abspath( str( target[0] ) ) )
-	if not os.path.isdir( dir ) :
-		os.makedirs( dir )
-
-	queryCommand = env["INKSCAPE"] + " --query-all \"" + svgFileName + "\""
-	objects = subprocess.check_output( queryCommand, shell=True ).decode()
-
-	for object in objects.split( "\n" ) :
-		tokens = object.split( "," )
-		if tokens[0].startswith( "forExport:" ) :
-			subprocess.check_call(
-				env["INKSCAPE"] + " --export-png=%s/%s.png --export-id=%s --export-width=%d --export-height=%d %s --export-background-opacity=0" % (
-					dir,
-					tokens[0].split( ":" )[-1],
-					tokens[0],
-					int( round( float( tokens[3] ) ) ), int( round( float( tokens[4] ) ) ),
-					svgFileName,
-				),
-				shell = True,
-			)
+def buildImageCommand( source, target, env ) :
+
+	# Requires env to have buildImageOptions set, containing, at minimum:
+	#	- id : The svg object id to export.
+
+	svgFilename = str( source[0] )
+	filename = str( target[0] )
+
+	substitutions = inkscapeArgs( env["buildImageOptions"], svgFilename )
+
+	outputDirectory = os.path.dirname( filename )
+	if not os.path.isdir( outputDirectory ) :
+		os.makedirs( outputDirectory )
+
+	args = " ".join( [
+			"--export-png={filePath}",
+			"--export-id={id}",
+			"--export-width={width:d}",
+			"--export-height={height:d}",
+			"--export-background-opacity=0",
+			"{svgPath}"
```

Comment:
Double indent

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/3711/commits/7c353686c74b1350f15fcad9123e848735eda703